### PR TITLE
Generic VPD parser

### PR DESCRIPTION
**Pull request by SunnySrivastava1984**

### CODE REVIEW

File: src/worker.cpp [hunk 1]
```diff
@@ -254,39 +249,11 @@ void Worker::fillVPDMap(const std::string& vpdFilePath,
...
-        }
-    }
-
     try
     {
-        std::shared_ptr<ParserInterface> parser =
-            ParserFactory::getParser(vpdVector, vpdFilePath, vpdStartOffset);
-        vpdMap = parser->parse();
+        std::shared_ptr<Parser> vpdParser =
+            std::make_shared<Parser>(vpdFilePath, m_parsedJson);
```
Comment (SunnySrivastava1984) [inline src/worker.cpp:255]: Make names more specific.

File: include/parser.hpp [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    /**
+     * @brief Constructor
+     *
+     * @param[in] vpdFilePath - Path to the VPD file.
+     * @param[in] parsedJson - Parsed JSON.
+     */
+    Parser(const std::string& vpdFilePath, nlohmann::json parsedJson);
+
+    /**
+     * @brief Api to implement a generic parsing logic.
```
Comment (SunnySrivastava1984) [inline include/parser.hpp:32]: Fix it and at other places as well.

File: include/utils.hpp [hunk 3]
```diff
@@ -261,5 +261,14 @@ void getVpdDataInVector(std::fstream& vpdFileStream,
 size_t getVPDOffset(const nlohmann::json& parsedJson,
                     const std::string& vpdFilePath);
 
+/**
+ * @brief API to parse respective JSON.
```
Comment (SunnySrivastava1984) [inline include/utils.hpp:265]: Fix JSON as Json.

File: src/utils.cpp [hunk 4]
```diff
@@ -449,5 +449,34 @@ size_t getVPDOffset(const nlohmann::json& parsedJson,
...
 }
+
+nlohmann::json getParsedJson(const std::string& pathToJson)
+{
+    if (pathToJson.empty())
+    {
+        throw std::runtime_error("Path to JSON is missing");
+    }
+
+    if (std::filesystem::exists(pathToJson) ||
```
Comment (SunnySrivastava1984) [inline src/utils.cpp:460]: Fix the condition. It should be "NOT"

File: src/parser_factory.cpp [hunk 5]
```diff
@@ -95,17 +95,20 @@ std::shared_ptr<ParserInterface>
     }
 
     vpdType type = vpdTypeCheck(vpdVector);
+    std::cout << vpdFilePath << std::endl;
```
Comment (SunnySrivastava1984) [inline src/parser_factory.cpp:98]: Have it with the parser selected log.

**Review by jinuthomas — APPROVED** [review]

**Pull request merged**